src/parser/jobs/drg/Procs.js

```diff
@@ -0,0 +1,153 @@
+import {Trans, Plural} from '@lingui/react'
+import React from 'react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+export default class Procs extends Module {
+	static handle = 'procs'
+	static dependencies = [
+		'downtime',
+		'suggestions',
+	]
+
+	_fangAndClaw = {
+		casts: 0,
+		removed: 0,
+	}
+	_wheelingThrust = {
+		casts: 0,
+		removed: 0,
+	}
+	_mirageDive = {
+		casts: 0,
+		removed: 0,
+		overwritten: 0,
+	}
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.FANG_AND_CLAW.id}, this._onFangAndClaw)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.WHEELING_THRUST.id}, this._onWheelingThrust)
+		this.addHook('cast', {by: 'player', abilityId: ACTIONS.MIRAGE_DIVE.id}, this._onMirageDive)
+		this.addHook('refreshbuff', {by: 'player', abilityId: STATUSES.DIVE_READY.id}, this._onReadyOverwritten) // The other two can't be overwritten due to how they drop
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.SHARPER_FANG_AND_CLAW.id}, this._onSharperRemoved)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.ENHANCED_WHEELING_THRUST.id}, this._onEnhancedRemoved)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.DIVE_READY.id}, this._onReadyRemoved)
+		this.addHook('complete', this._onComplete)
+	}
+
+	// For all of our cast/removal tracking, we only want to know if it happened outside of downtime to avoid errant penalization.
+	// The only one that doesn't need the check is overwriting Mirage Dive, since you can't get a proc from an invulnerable target.
+	_onFangAndClaw(event) {
```

Comment:
Could probably combine a few of these with a higher order function or just more general event hooking, unless you'd expect that some would need special handling in some manner.

Reply:
Nah, none of them need anything special. I can twiddle it so the cast/remove events are generalized and make one object that keys on ability ID to wrap the three separate ones I have now.